### PR TITLE
Add `__len__` and `__delitem__` to Scene

### PR DESCRIPTION
**Pull request by tiffanyxiao**
For Issue[https://github.com/pythonarcade/arcade/issues/1651](https://github.com/pythonarcade/arcade/issues/1651)

## What was done

- Added `__len__` and `__delitem__` methods to `scene.py`. 
- Added tests for these new scene methods.

## How to test
- Run the pytests; both tests should pass (the individual tests are in `tests/unit/scene/test_scene_len_delitem.py`).

Special thanks to @Cleptomania for the hands on help at PyCon 2023!

### CODE REVIEW

File: arcade/scene.py [hunk 1]
```diff
@@ -29,6 +29,24 @@ def __init__(self) -> None:
...
+
+    def __delitem__(self, sprite_list: Union[str, SpriteList]) -> None:
+        """
+        Remove the SpriteList from `_sprite_lists` and `_name_mapping`.
+
+        :param Union[str, SpriteList] sprite_list: which SpriteList to delete - can
+        be either the name of the SpriteList or the SpriteList object.
+        """
+        if isinstance(sprite_list, str):
+            self.remove_sprite_list_by_name(sprite_list)
```
Comment (gran4) [inline arcade/scene.py:46]: Maybe you could add an is instance check for an integer, because I'm sure some users will want that. Like: a animal simulator.

Comment (gran4) [reply arcade/scene.py:46]: Maybe you could make a self.remove_sprite_list_by_index.

**Review by gran4 — COMMENTED** [review]
Other than that it looks good to me.

File: arcade/scene.py (hunk 1 above)
Comment (gran4) [reply arcade/scene.py:46]: Too late I guess now.

Comment (Cleptomania) [reply arcade/scene.py:46]: I would definitely be open to a follow up PR adding a case for integers to remove based on the list index.

Comment (gran4) [reply arcade/scene.py:46]: @tiffanyxiao do you want to do this? If you don't I'll do it.

Comment (tiffanyxiao) [reply arcade/scene.py:46]: Sure, I'll try it!

**Pull request merged**